diff --git a/cpp/src/arrow/flight/CMakeLists.txt b/cpp/src/arrow/flight/CMakeLists.txt
index d0f4f3cf6..446437581 100644
--- a/cpp/src/arrow/flight/CMakeLists.txt
+++ b/cpp/src/arrow/flight/CMakeLists.txt
@@ -195,3 +195,11 @@ if(ARROW_BUILD_BENCHMARKS)
 
   add_dependencies(arrow_flight arrow-flight-benchmark)
 endif(ARROW_BUILD_BENCHMARKS)
+
+add_executable(send-to-dest send_to_dest.cc common.cc in_memory_storage.cc sender.cc)
+target_link_libraries(send-to-dest ${ARROW_FLIGHT_TEST_LINK_LIBS}
+                      ${GFLAGS_LIBRARIES} GTest::GTest plasma_shared)
+
+add_executable(receive-and-store receive_and_store.cc common.cc in_memory_storage.cc)
+target_link_libraries(receive-and-store ${ARROW_FLIGHT_TEST_LINK_LIBS}
+                      ${GFLAGS_LIBRARIES} GTest::GTest plasma_shared)
