diff --git a/cpp/src/arrow/flight/CMakeLists.txt b/cpp/src/arrow/flight/CMakeLists.txt
index d0f4f3cf6..84d26786b 100644
--- a/cpp/src/arrow/flight/CMakeLists.txt
+++ b/cpp/src/arrow/flight/CMakeLists.txt
@@ -195,3 +195,11 @@ if(ARROW_BUILD_BENCHMARKS)
 
   add_dependencies(arrow_flight arrow-flight-benchmark)
 endif(ARROW_BUILD_BENCHMARKS)
+
+add_executable(sort-and-send sort_and_send.cc sorter.cc sender.cc)
+target_link_libraries(sort-and-send ${ARROW_FLIGHT_TEST_LINK_LIBS}
+                      ${GFLAGS_LIBRARIES} GTest::GTest)
+
+add_executable(receive-and-merge receive_and_merge.cc in_memory_storage.cc)
+target_link_libraries(receive-and-merge ${ARROW_FLIGHT_TEST_LINK_LIBS}
+                      ${GFLAGS_LIBRARIES} GTest::GTest plasma_shared)
